packages/pytest-simcore/src/pytest_simcore/simcore_storage_service.py

```diff
@@ -45,22 +45,24 @@ async def storage_service(
 ) -> URL:
     await wait_till_storage_responsive(storage_endpoint)
 
-    def correct_ip(url: AnyUrl):
+    def _replace_storage_endpoint(url: str) -> str:
+        url_obj = TypeAdapter(AnyUrl).validate_python(url)
         assert storage_endpoint.host is not None
         assert storage_endpoint.port is not None
 
-        return AnyUrl.build(
-            scheme=url.scheme,
+        storage_endpoint_url = AnyUrl.build(
+            scheme=url_obj.scheme,
             host=storage_endpoint.host,
-            port=f"{storage_endpoint.port}",
-            path=url.path,
-            query=url.query,
+            port=storage_endpoint.port,
+            path=url_obj.path.lstrip("/"),
+            query=url_obj.query,
         )
+        return f"{storage_endpoint_url}"

```

Comment:
any option to use URL from yarl and do something to change the port? at this point this looks very cumbersome.